### PR TITLE
AP_RTC: add local time support

### PR DESCRIPTION
**Pull request by vierfuffzig**
adds timezone offset [+- hrs] to calculate local time from UTC

### CODE REVIEW

**Comment by peterbarker** [comment]
Need finer resolution than this.  Not every timezone is aligned on a 1-hour boundary.  I do NOT know how fine is required - 30m at least.

**Comment by vierfuffzig** [comment]
thanks @peterbarker . actually i wasn't aware of sub-hour timezone offsets. couldn't find offsets less than quarter of an hour, so i changed to minutes

**Comment by peterbarker** [comment]
@vierfuffzig while I haven't looked deeply at the implementation here (looks solid at first glance), we have a policy of not carrying dead code in ArduPilot.  Most of the time we do actually manage to stick to that rule.

At this point there are no callers to the new methods, which makes this dead code.

Do you have a use-case in mind for the new code?  The only thing I can think of is delay-until-local-time-of-day - but there's obviously none of that in here.

**Comment by peterbarker** [comment]
... *sigh*.  And *then* see your OSD PR @ https://github.com/ArduPilot/ardupilot/pull/12495

**Comment by vierfuffzig** [comment]
sorry @peterbarker i should have been clearer about this PR's purpose. on a sidenote i just managed to mess this up trying to squash commits, but i think i got it straightened out again...

File: libraries/AP_RTC/AP_RTC.cpp [hunk 1]
```diff
@@ -27,6 +27,13 @@ const AP_Param::GroupInfo AP_RTC::var_info[] = {
     // @Bitmask: 0:GPS,1:MAVLINK_SYSTEM_TIME,2:HW
     // @User: Advanced
     AP_GROUPINFO("_TYPES",  1, AP_RTC, allowed_types, 1),
+    
+    // @Param: _TZ
```
Comment (tridge) [inline libraries/AP_RTC/AP_RTC.cpp]: I think calling it _TZ_MIN would be clearer

**Comment by vierfuffzig** [comment]
closing this, moving it into OSD time panel PR to make CI pass as @tridge suggested. thanks everyone.

**Pull request closed without merging**